presto-main/src/main/resources/webapp/src/components/Splits.jsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Timeline, DataSet } from "vis-timeline/standalone";
+import { useRef, useEffect, useState } from 'react';
+import { getFirstParameter } from "../utils";
+import { QueryHeader } from "./QueryHeader";
+
+
+export default function Split(): void {
+
+    const containerRef = useRef(null);
+    const timelineRef = useRef(null);
+    const timerid = useRef(0);
+    const [queryState, setQueryState] = useState({query: null, failed: false, ended: false, items: null, groups: null});
+
+    function calculateItemsGroups(query) {
+        var getTasks = (stage) => {
+            return [].concat.apply(
+                    stage.latestAttemptExecutionInfo.tasks,
+                    stage.subStages.map(getTasks));
+        }
+        var tasks = getTasks(query.outputStage);
+        tasks = tasks.map((task) => {
+            return {
+                taskId: task.taskId.substring(task.taskId.indexOf('.') + 1),
+                time: {
+                    create: task.stats.createTime,
+                    firstStart: task.stats.firstStartTime,
+                    lastStart: task.stats.lastStartTime,
+                    lastEnd: task.stats.lastEndTime,
+                    end: task.stats.endTime,
+                },
+            };
+        });
+
+        var groups = new DataSet();
+        var items = new DataSet();
+        for (var i = 0; i < tasks.length; i++) {
```

Comment:
In this case, a `le` should work
```suggestion
        for (let i = 0; i < tasks.length; i++) {
```